src/object/obj_tx.c

```diff
@@ -3254,6 +3254,18 @@ dc_tx_convert_post(struct dc_tx *tx, tse_task_t *task, enum obj_rpc_opc opc, int
 		goto out;
 	}
 
+	/*
+	 * Original task has ever been scheduled, and is required to be handled via distributed
+	 * transaction. Let's re-schedule it.
+	 */
+	if (task->dt_result != 0) {
```

Comment:
any failure will resched it even for unretryable failure? will it cause dead loop for some case? (for example EC 4+2 update task hit 3 shard failure).

Reply:
It may be "DER_NEED_TX" that is used to trigger dc_tx_convert().

Reply:
See ASSERT check in dc_tx_convert(), we only allow either zero or DER_NEED_TX

Reply:
ok, seems a problem that when enter dc_tx_convert() the task->dt_result is either 0 or -DER_NEED_TX, but here in  dc_tx_convert_post() the dt_result possible changed? for example in dc_tx_attach() it may created and scheduled check_existence_task dep task that may affect task->dt_result?

Reply:
>  for example in dc_tx_attach() it may created and scheduled check_existence_task dep task that may affect task->dt_result?

That is possible. But if the logic comes here, that means related TX can be committed, then it will create the commit dependency task. Otherwise, if it is other task->dt_result, it will "goto out" to complete the task directly.

Reply:
ok, thx. Or maybe here L3261 can change to "if (task->dt_result == -DER_NEED_TX)" is clearer?